File: 20170206_CustomExceptions/CustomExceptions.py
class MyException(Exception):
    # inherit from base class Exception
    def __init__(self, err):
        self.errors = err


if __name__ == "__main__":
    try:
        # force exception to occur
        raise MyException("bad bad error")   
    except MyException as e:
        print("except block")
        print(e.errors)
       
      

File: 20170206_CustomExceptions/CustomExceptions.java
public class CustomExceptions{
  /**
  *Implement MyException created in the file called
  * MyException.java.
  */
  static void testMethod(int x) throws MyException{
    if(x < 0)
      throw new MyException("bad bad error");
    else
      System.out.print("input valid");
  }

  public static void main(String[] args){
    // use a try catch statement to catch the 
    // exception defined in testMethod
    try {
       testMethod(-14);
    }
    catch (MyException e) {
      e.printStackTrace();
    }
  }
}
